libs/live-signer-solana/src/DMKSignerSolana.ts

```diff
@@ -0,0 +1,179 @@
+            reject(this._mapError<GetAppConfigurationDAError>(state.error));
+          }
+          if (state.status === DeviceActionStatus.Completed) {
+            const { version, blindSigningEnabled, pubKeyDisplayMode } = state.output;
+            const mode =
+              pubKeyDisplayMode === this.DMKPubKeyDisplayMode.long
+                ? PubKeyDisplayMode.LONG
+                : PubKeyDisplayMode.SHORT;
+            resolve({ version, blindSigningEnabled, pubKeyDisplayMode: mode });
+          }
+        },
+        error: err => {
+          reject(err);
+        },
+      });
+    });
+  }
+
+  /**
+   * retrieves a Solana address for the given derivation path.
+   * @param path - BIP32 derivation path
+   * @param display - whether to prompt user confirmation on device
+   */
+  async getAddress(path: string, display?: boolean): Promise<SolanaAddress> {
+    console.log("Fetching address for path via DMK");
+    const { observable } = this.dmkSigner.getAddress(path, { checkOnDevice: !!display });
+    return new Promise<SolanaAddress>((resolve, reject) => {
+      observable.subscribe({
+        next: state => {
+          if (state.status === DeviceActionStatus.Error) {
+            reject(this._mapError<GetAddressDAError>(state.error));
+          }
+          if (state.status === DeviceActionStatus.Completed) {
+            const addressString = state.output;
+            const addressBytes = Buffer.from(bs58.decode(addressString));
+            resolve({ address: addressBytes });
+          }
+        },
+        error: err => {
+          reject(err);
+        },
+      });
+    });
+  }
+
+  /**
+   * signs a Solana transaction via DMK.
+   * @param path - BIP32 derivation path
+   * @param txBuffer - transaction data as a uint8 array
+   */
+  async signTransaction(path: string, txBuffer: Uint8Array): Promise<SolanaSignature> {
+    console.log("Signing transaction via DMK");
+    const { observable } = this.dmkSigner.signTransaction(path, txBuffer);
+    return new Promise<SolanaSignature>((resolve, reject) => {
+      observable.subscribe({
+        next: state => {
+          if (state.status === DeviceActionStatus.Error) {
+            reject(this._mapError<SignTransactionDAError>(state.error));
+          }
+          if (state.status === DeviceActionStatus.Completed) {
+            const signatureBuffer = Buffer.from(state.output);
+            resolve({ signature: signatureBuffer });
+          }
+        },
+        error: err => {
+          reject(err);
+        },
+      });
+    });
+  }
+
+  /**
+   * signs an offchain message displayed on device via DMK.
+   * @param path - BIP32 derivation path
+   * @param messageHex - message to sign in hexadecimal format
+   */
+  async signMessage(path: string, messageHex: string): Promise<SolanaSignature> {
+    console.log("Signing message via DMK");
+    const { observable } = this.dmkSigner.signMessage(path, messageHex);
```

Comment:
I've also checked the code of the dmk signer for solana, and I think you are missing some logic to find the correct message format if it's ASCII or UTF-8 strings and depending on the length also you can get a different format with the Sol app

https://github.com/LedgerHQ/ledger-live/blob/develop/libs/coin-modules/coin-solana/src/offchainMessage/format.ts#L78 vs https://github.com/LedgerHQ/device-sdk-ts/blob/develop/packages/signer/signer-solana/src/internal/app-binder/task/SendSignMessageTask.ts#L105

Reply:
this part should be fixed on the signer kit directly, not related to this PR I guess